### PR TITLE
Add fetch-file and download file function, add file setting modal

### LINKED ISSUE #18: File setting modal
Each file card has a button on the right side which doesn't do anything yet:
image

When clicknig it a modal should open at the same position that has the following options:
- delete
- download

If other options come to mind please add them in the comments.

### LINKED ISSUE #23: Create function for downloading files on to the device
We need a function with which the user can download a file to his/her device.

### PR DESCRIPTION
**Pull request by theopfr**
Closes #18 
Closes #23

The goal of this MR is:
- [x] fix image preview on file cards
- [x] add a modal popup to the file cards
- [x] add file-download to file settings modal
- [x] add file-deletion to file settings modal

I added two util funtions: ``fetchFile`` and ``downloadFile``. I already started to integrate them into the ``FileCard`` component so now you can see image previews on them. Also, on each card there are these three dots on which the user should be able to download/delete the file.
Currently, clicking on the three dots just simply downloads the file. Due to my phone being weird I cannot 100% test if the download works (my file app just isnt loading, dont know why).

So please check if the download works for you. Maybe also someone should check on iOS.

Depends on MR #22 
~This MR depends on another which will add the util function for deleting files.~

### CODE REVIEW

**Comment by cma5** [comment]
It seems to download the file i selected. The only problem is, that it doesn't put it on the phones drive.
Unbenannt

**Comment by cma5** [comment]
You also have to make a new folder in the dialog in order to set a saving location. But as I said before, the file isn't actually saved there

grafik

**Comment by cma5** [comment]
And on IOs like you said it doesn't work anyway.

**Comment by cma5** [comment]
grafik
2601E1E1-D47B-4C6E-98A4-484834C395FB
Functionality wise everything is great, the only thing I would change ist to remove the popup on IOs since when you move out of the share dialog without saving, the file has been downloaded doesn't make sense here.

File: app/utils/ServerRequests.tsx [hunk 1]
```diff
@@ -144,28 +144,122 @@ export const fetchFile = async (fileURL: string): Promise<string | void> => {
...
+            const fileUri = `${FileSystem.cacheDirectory}${fileName}`;
+
+            await FileSystem.writeAsStringAsync(fileUri, base64Data.split(",")[1], {
+                encoding: FileSystem.EncodingType.Base64,
+            });
+
+            await Sharing.shareAsync(fileUri, { mimeType: "application/octet-stream", dialogTitle: "Share the file" });
+        }
+
+        Alert.alert("File Downloaded", `File has been downloaded.`);
```
Comment (cma5) [inline app/utils/ServerRequests.tsx]: please put this only into the android thingie

**Review by cma5 — CHANGES_REQUESTED** [review]

File: app/utils/ServerRequests.tsx (hunk 1 above)
Comment (theopfr) [reply app/utils/ServerRequests.tsx]: The alert text? Can't iOS phones get alerts? I replaced the download ocnfirmation with a toast anyways, but for failures there still is an Alert

**Review by cma5 — APPROVED** [review]
LGTM 👍

**Pull request merged**